### DISCUSSION TITLE
Check if all required fields are filled in

### DISCUSSION THREAD

**Discussion by IgnusG** [react-hook-form/react-hook-form#8881] [Q&A] [answered]
I'd like to implement a design where the submit button is disabled until all the required fields are filled in. I'd like it to get enabled once all the required fields are dirty - not necessarily fully valid though (eg. the length of an input might not be satisfied yet etc.).

I can't use the `isValid` state because I just want to check for required field "validations" and some of my fields might have additional validation rules that don't need to be satisfied for the button to be enabled.

I'm not sure if there's any state or method I can use to defer this to react-hook-form or if I have to somehow loop over the fields/validations and manually check them? Maybe someone knows or has tried to solve something similar before?

**Comment by bluebill1049** [comment, marked as answer]
> I can't use the isValid state because I just want to check for required field "validations" and some of my fields might have additional validation rules that don't need to be satisfied for the button to be enabled.

maybe consider building a custom resolver for this and checking `isValid` state?

**Reply by IgnusG** [reply]
Yeah I’ve discovered custom resolvers after writing this and was thinking about using them to do the job. Not the most out of the box experience but it is a very specific behaviour I’m looking for after all 😅

I’ll have to probably extract the required field information out of the inputs since the native `isValid` state still checks other conditions too (for all fields as well I think).

Thanks for the tip tho!
